feature/electron/spi1-dma-bugfix

avtolstoy opened the pull request:
`DMA1_Stream2_IRQn` used by SPI1 and SPI2 on Electron should be handled by `DMA1_Stream2_irq()`

Fixes issue #1080 

---

Doneness:

- [X] Contributor has signed CLA
- [X] Problem and Solution clearly stated
- [x] Code peer reviewed
- [x] API tests compiled
- [x] Run unit/integration/application tests on device
- [x] Add documentation (N/A)
- [ ] Add to CHANGELOG.md after merging (add links to docs and issues)

### BUGFIX

- [Electron] `DMA1_Stream2_IRQn` used by SPI1 and SPI2 on Electron should be handled by `DMA1_Stream2_irq()`. Fixes [#1080](https://github.com/spark/firmware/issues/1080)

technobly commented:
@avtolstoy would you please add SPI1 and SPI2 to the SPI master/slave tests?  I think that will be a nice validation that everything is working well on the Electron now.

avtolstoy commented:
@technobly 
- Additional tests in `wiring/no_fixture`:
  - `SPI_02_SPI_DMA_Transfers_Work_Correctly`
  - `SPI_03_SPI1_DMA_Transfers_Work_Correctly`
  - `SPI_04_SPI2_DMA_Transfers_Work_Correctly`
- Added `USE_SPI` variable for `spi_master_slave` tests to switch between `SPI/SPI1/SPI2`, e.g. `make USE_SPI=SPI2`
- Added wiring diagrams in `spi_master_slave` :)
